tests/real_tests.rs

```diff
@@ -75,114 +83,161 @@ fn get_devices(count: u8) -> Option<Vec<RealTestDev>> {
         .unwrap()
         .as_array()
         .unwrap();
-    if devpaths.len() < count as usize {
-        return None;
+
```

Comment:
Having DeviceLimits be a single struct, as:

```
struct DeviceLimits {
  lower: usize, // inclusive
  upper: Option<usize> // exclusive 
}

// impl contains a new that ensures lower < upper if upper not None.
```

would work just as well. 

```DeviceLimits { lower: 9, upper: Some(10) }``` means Exactly(9)
```DeviceLimits { lower: 9, upper: None }``` means AtLeast(9)
```DeviceLimits { lower: 9, upper: Some(30) }``` means Range(9, 29)

Then,

```
let avail = devpaths.len();
if limits.lower > avail {
  return None;
}
let high_take = {
  if limits.lower == avail {
    None
  } else {
    match limits.upper {
      None => Some(avail),
      Some(upper) => {
        if limits.lower + 1 == upper {
          None
        } else {
          Some(min(limits.upper - 1, avail))
        }
      }
    }
  }
}
```
```low_take``` is just ```limits.lower``` and continue as before.